### PR TITLE
Actionification

### PR DESCRIPTION
**Pull request by Maciej4**
This adds a new object, Trigger.java, that should allow for the addition of actions to the parallel scheduler on the press of a button. It also includes two tests.

### CODE REVIEW

File: src/main/java/frc/robot/actions/VisionAlign.java [hunk 1]
```diff
@@ -33,10 +32,32 @@
...
+        timeoutCounter = 0.0;
+
+        headingPID = new PID(headingP, headingI, headingD);
+        
+        this.startTime = System.currentTimeMillis();
+    }
 
     public void loop()
     {
+        //TODO: Combine track cancelling into action
```
Comment (HoldenA1) [inline src/main/java/frc/robot/actions/VisionAlign.java:46]: There is a lot of commented code here. The main branch should be only working code. If the code is incomplete, it probably shouldn't be merged yet. If this part is being worked on in another branch it should be merged there first.

File: src/main/java/frc/robot/util/Trigger.java [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    public Trigger (int buttonID, Action action) {
+        this.buttonID = buttonID;
+        this.action = action;
+        gson = new Gson();
+    }
+
+    public void loop()
+    {
+        if(Context.robotController.driverJoystick.getButtonPressed(buttonID)){
+            String seralized = gson.toJson(this.action);
```
Comment (HoldenA1) [inline src/main/java/frc/robot/util/Trigger.java:24]: It isn't obvious what a google library for converting stuff to JSON is doing here. This a place where comments would be nice to have.

File: src/test/java/frc/robot/controllers/DriverJoystickTest.java [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+import frc.robot.execution.ParallelScheduler;
+import frc.robot.util.Context;
+import frc.robot.util.Trigger;
+
+public class DriverJoystickTest {
+    public DriverJoystick driverJoystick;
+
+    public class Complete extends Action{
+        public void loop()
+        {
```
Comment (HoldenA1) [inline src/test/java/frc/robot/controllers/DriverJoystickTest.java]: While code style really isn't that big of an issue, the curly braces are very inconsistent. The opening curly braces should all be on the same line as the function declaration. This should not change for if statements, for loops, while loops, or class declarations. It just has to be consistent.

**Review by HoldenA1 — CHANGES_REQUESTED** [review]
Bulk of the code looks fine, but some things should be addressed

**Comment by Maciej4** [comment]
I think that I have fixed all of the issues you have posed.

**Review by HoldenA1 — DISMISSED** [review]
Looks good now.

**Comment by Maciej4** [comment]
Fixed merge conflicts, which merged into branch and dismissed review. Sorry about that, could you re-review.

**Review by HoldenA1 — APPROVED** [review]
Rip github

**Pull request merged**